dialect/pgdialect/array.go

```diff
@@ -389,10 +389,6 @@ func arrayScanner(typ reflect.Type) schema.ScannerFunc {
 			}
 		}
 
-		if src == nil {
-			return nil
-		}
```

Comment:
This is array, not range. If you think this should be changed, let's do it in a separate PR ensuring that we have a test for a NULL array.